### ISSUE TITLE
No data sent to Coveralls casued by absolute paths in cobertura file

### ISSUE THREAD

**Issue by Martomate** [scoverage/sbt-coveralls#199]
### Environment
Scala 3.2.0, sbt-scoverage 2.0.2, sbt-coveralls 1.3.2

### Problem
No coverage data is sent to Coveralls

### Possible cause
The cobertura.xml file contains absolute paths for the class filenames, which results in sbt-coveralls not finding any source files.

This happens on this line, where the `relativePath` is not relative: https://github.com/scoverage/sbt-coveralls/blob/8c6466572d3a8dc4112e80d5c63cce52ee0a9d8c/src/main/scala/org/scoverage/coveralls/CoberturaMultiSourceReader.scala#L54

This is likely caused by the new scoverage version for Scala 3, where there were some changes to the cobertura.xml file. It now starts like this:

```xml
<?xml version="1.0"?>
<!DOCTYPE coverage SYSTEM "http://cobertura.sourceforge.net/xml/coverage-04.dtd">
<coverage 
line-rate="0.29" lines-valid="7499" lines-covered="2198" branches-valid="593" branches-covered="175" branch-rate="0.30" complexity="0" version="1.0" timestamp="1662191703030">
    <sources>
        <source>--source</source>
        <source>/Users/martin/git/myproject/src/main/scala</source>
    </sources>
    <packages>
        <package name="ai" line-rate="0.00" branch-rate="0.00" complexity="0">
            <classes>
                <class 
                name="a.MyClass" filename="/Users/martin/git/myproject/src/main/scala/a/MyClass.scala" line-rate="0.00" branch-rate="0.00" complexity="0">
                    <methods>
                        <method name="a/MyClass/myFunction" signature="()V" line-rate="0.00" branch-rate="0.00" complexity="0">
                            <lines>
                                <line number="17" hits="0" branch="false"/>
```

**Comment by rolandtritsch** [comment]
Hi @Martomate. I can reproduce this.

Hi @ckipp01. This looks like a bug. If it is ok with you, I will take a look at this. I think these paths should be relative (like the paths in all the other output files).

**Comment by ckipp01** [comment]
> Hi @ckipp01. This looks like a bug. If it is ok with you, I will take a look at this. I think these paths should be relative (like the paths in all the other output files).

For sure!

**Comment by rolandtritsch** [comment]
Proposed fix - https://github.com/scoverage/scalac-scoverage-plugin/pull/497

**Comment by rolandtritsch** [comment]
@ckipp01 After this/the fix is merged and we have a new version of `scalac-scoverage-plugin` we need to rebuild/publish a new version of `sbt-coverage`.

**Comment by ckipp01** [comment]
> @ckipp01 After this/the fix is merged and we have a new version of scalac-scoverage-plugin we need to rebuild/publish a new version of sbt-coverage.

2.12.17 and 2.13.9 are scheduled for release this week, so I'll probably wait for those, and then cut the release which will include the fix.

**Comment by Martomate** [comment]
This seems to have been fixed in sbt-scoverage 2.0.4, and I can confirm that it now works! Thanks a lot for fixing this so quickly!

**Comment by Falmarri** [comment]
Why was the fix for this to make this relative, rather than just use the existing absolute file path? Right now this is a heuristic to find the file. There can be multiple file names that have the same relative path in the source directories, especially on aggregated builds between subprojects. 

https://github.com/scoverage/scalac-scoverage-plugin/issues/582

It seems absurd to not just check to see if the path is absolute and use that directly rather than try to search through the list of source directories to find a path that matches.
